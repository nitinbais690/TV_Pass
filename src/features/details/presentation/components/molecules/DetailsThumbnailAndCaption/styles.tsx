import { scale } from 'qp-common-ui';
import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    contentDetails: {
        flexDirection: 'row',
        marginHorizontal: scale(16),
    },
    caption: {
        flex: 1,
    },
});
