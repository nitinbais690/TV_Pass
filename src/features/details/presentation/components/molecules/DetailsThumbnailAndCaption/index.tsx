import React from 'react';
import { View } from 'react-native';
import { styles } from './styles';
import DetailsTitleAndCaption from '../../molecules/TitleAndCaption';
import DetailsThumbnail from '../../atoms/DetailsThumbnail';

export default function DetailsThumbnailAndCaption(props: DetailsThumbnailAndCaptionProps) {
    return (
        <View style={styles.contentDetails}>
            <DetailsThumbnail id={props.resourceId || ''} mainResourceImage={props.image} imageUrl={props.imageUrl} />
            <View style={styles.caption}>
                <DetailsTitleAndCaption
                    seriesTitle={props.seriesTitle}
                    title={props.name}
                    caption={props.caption1String}
                    advisory={props.advisory}
                />
            </View>
        </View>
    );
}

export interface DetailsThumbnailAndCaptionProps {
    advisory: string | undefined;
    caption1String: string | undefined;
    image: string | undefined;
    resourceId: string | undefined;
    seriesTitle: string | undefined;
    imageUrl: string | undefined;
    name: string | undefined;
}
