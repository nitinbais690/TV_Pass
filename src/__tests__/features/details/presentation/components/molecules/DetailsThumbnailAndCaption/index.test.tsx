import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import DetailsThumbnailAndCaption from 'features/details/presentation/components/molecules/DetailsThumbnailAndCaption';

jest.mock('core/presentation/hooks/use-app-colors');

describe('detailsThumbnailAndCaption', () => {
    const props = {
        image: 'assets/images/video_placeholder.png',
        resourceId: '01234',
        imageUrl: 'assets/images/got.png',
        seriesTitle: 'Game of Thrones',
        name: 'Season 1: Winter Is Coming',
        caption1String: '2011 . 50 Minutes . Drama',
        advisory: 'violence',
    };

    test('renders correctly', () => {
        const { container } = render(
            <DetailsThumbnailAndCaption
                resourceId={props.resourceId}
                image={props.image}
                imageUrl={props.imageUrl}
                advisory={props.advisory}
                seriesTitle={props.seriesTitle}
                name={props.name}
                caption1String={props.caption1String}
            />,
        );
        expect(toJSON(container)).toMatchSnapshot();
    });
});
