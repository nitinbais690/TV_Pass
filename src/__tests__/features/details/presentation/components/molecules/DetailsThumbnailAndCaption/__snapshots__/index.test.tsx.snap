// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detailsThumbnailAndCaption renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginHorizontal": 23.314285714285713,
      }
    }
  >
    <View
      style={
        Object {
          "height": 104.91428571428571,
          "width": 90.34285714285714,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "aspectRatio": 0.6666666666666666,
                "borderRadius": 11.657142857142857,
                "flex": 1,
                "width": "100%",
              },
            ],
          ]
        }
      >
        <FastImageView
          resizeMode="contain"
          source={
            Object {
              "cache": "web",
              "uri": "assets/images/got.png",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <Text
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontFamily": "ProximaNova-Black",
                "fontSize": 40,
                "fontWeight": "900",
              },
              Object {
                "color": "#ECECEC",
                "fontFamily": "ProximaNova-Black",
                "fontSize": 28,
                "fontWeight": "900",
                "textTransform": "uppercase",
              },
              undefined,
            ]
          }
        >
          Game of Thrones
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "ProximaNova-Regular",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "color": "#FFFFFF",
                "fontFamily": "ProximaNova-Semibold",
                "opacity": 0.3,
              },
              Object {
                "marginTop": 23.314285714285713,
              },
            ]
          }
        >
          2011 . 50 Minutes . Drama
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "ProximaNova-Regular",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "color": "#FFFFFF",
                "fontFamily": "ProximaNova-Semibold",
                "opacity": 0.3,
              },
              undefined,
            ]
          }
        >
          violence
        </Text>
      </View>
    </View>
  </View>
</View>
`;
